src/destreamer.ts

```diff
@@ -274,5 +218,6 @@ async function main() {
     downloadVideo(videoUrls, argv.outputDirectory, session);
 }
 
-
+var tokenCache = new TokenCache();
```

Comment:
this is OT, just a tip

var is "deprecated" because it has no scope, it can cause a lot of headache and nasty bugs
there are some very rare occasions where var could help you (because it has no scope) but usually it means you should rewrite that code :P

Reply:
OK thanks for the tip since my research online only came to the conclusion that var, let and const have different scopes and that const is indeed constant :P

Reply:
var i=0 is inherited everywhere in your program, even in a for loop with its own var i=0 (that means you loose your loop variable)
it was a mess before let and const and the memory usage was huge

let and const follow the standard rules, use const everywhere and let only if you need to assign that variable later on ;)